src/main/java/randoop/generation/Bloodhound.java

```diff
@@ -0,0 +1,237 @@
+
+    return operation;
+  }
+
+  /**
+   * The branch coverage information for all methods under test is re-summarized at every {@code
+   * branchCoverageInterval}'th call of this method. Weights for all methods under test are
+   * recomputed when branch information is re-summarized.
+   */
+  private void updateBranchCoverageMaybe() {
+    stepNum += 1;
+    if (stepNum % branchCoverageInterval == 0) {
+      methodSelectionCounts.clear();
+      CoverageTracker.instance.summarizeCoverageInformation();
+      updateWeightsForAllOperations();
+    }
+  }
+
+  /** Computes and updates weights in our method weights map for all methods under test. */
+  private void updateWeightsForAllOperations() {
+    for (TypedOperation operation : operationSimpleList) {
+      updateWeightForOperation(operation);
+    }
+  }
+
+  /**
+   * Recompute weights for a method under test. A method under test is assigned a weight based on a
+   * weighted combination of the number of branches uncovered and the ratio between the number of
+   * times this method has been selected and the maximum number of times any method under test has
+   * been selected. The weighting scheme is based on Bloodhound in the Guided Random Testing (GRT)
+   * paper.
+   *
+   * @param operation method to compute weight for
+   */
+  private void updateWeightForOperation(TypedOperation operation) {
+    // The number of methods under test, corresponds to |M| in the GRT paper.
+    int numOperations = this.operationSimpleList.size();
+
+    // Default weight is uniform probability.  It is used if no coverage details are available.
+    // This is the case for classes like java.lang.Object which is not explicitly under test.
+    double weight = 1.0 / numOperations;
+
+    CoverageTracker.BranchCoverage covDet =
+        CoverageTracker.instance.getDetailsForMethod(operation.getName());
+
+    // Check that branch coverage details are available for this method.
+    if (covDet != null) {
+      // The number of successful invocations of this method. Corresponds to succ(m).
+      Integer numSuccessfulInvocations = methodInvocationCounts.get(operation);
+      if (numSuccessfulInvocations == null) {
+        numSuccessfulInvocations = 0;
+      }
+
+      // If this method has no uncovered branches, or it is the most-selected method,
+      // use the default weight and skip this step.
+      if (covDet.uncoveredBranches != 0
+          && numSuccessfulInvocations != maxSuccessfulInvocationsOfAnyMethod) {
+        // Uncovered branch ratio of this method. The name uncovRatio(m) is from the GRT paper.
+        double uncovRatio;
+        if (covDet.totalBranches == 0) {
+          uncovRatio = 0.5;
+        } else {
+          uncovRatio = (double) covDet.uncoveredBranches / covDet.totalBranches;
+        }
+
+        // Call ratio of this method. Corresponds to succ(m) / maxSucc(M) in the GRT paper.
+        double callRatio;
+
+        if (maxSuccessfulInvocationsOfAnyMethod == 0) {
```

Comment:
I suggest initializing `maxSuccessfulInvocationsOfAnyMethod` to 1, in order to simplify this code.  (It won't need the check any more.)  You can document at its initialization that this is because it appears in the denominator of a division.